### ISSUE TITLE
artery - Rolling update from 2.5.11 to 2.5.12 - excessive negative acknowledgement

### ISSUE THREAD

**Issue by marekzebrowski** [akka/akka#25143]
I'm trying to perform rolling update from system using 2.5.11 to 2.5.12
system  uses artery-udp for transport.

shortly after restart I can observe in logs thousands of entries:
```
a.r.a.c.InboundActorRefCompression: Inbound message from originUid [6452111113621292620] is using unknown compression table version. It may have
been sent with compression table built for previous incarnation of this system. Versions activeTable: 0, nextTable: 1, incomingTable: 102
```
shortly after few thousands of others
```
a.r.a.SystemMessageAcker: Sending negative acknowledgement of system message [16] from [..address here...] , highest acknowledged [0]
```

only full restart with downtime helps in that situation, cluster cannot recover from that situation

**Comment by patriknw** [comment]
It's indeed interesting to understand what went wrong here. I tried to reproduce with [akka-sample-sharding-scala](https://github.com/akka/akka-samples/tree/2.5/akka-sample-sharding-scala). Picked that one because sharding is using most cluster features and `watch` (system messages).

I have been able to reproduce the negative ack, but not the compression issue.

1. Started 3 nodes, 2551, 2552, 2553 with 2.5.11.
1. Changed to 2.5.12, and restarted 2553 with ctrl-c (graceful leaving). No problem.
1. Restarted 2552 with kill -9, and then it started to log the negative ack in 2552

I think we fixed several things in this version, which were kind of silently broken previously. Perhaps this would solve itself when all have been restarted to the new version and in the meanwhile one would have to accept the logging, or add a logging filter.

I'll do some more testing.

**Comment by patriknw** [comment]
ah, I had the compression issue also in the logs

**Comment by patriknw** [comment]
It actually continues to send the negative acks after restarting 2511 also. That was surprising, and indication of a serious problem.

I also tried with a rolling restart with same version, 2.5.12, and then it works fine.

By the way, the compression warning is probably not the issue here. It is as it says "sent with compression table built for previous incarnation of this system", which is expected and will heal itself shortly.

Anyway, seems like we have a good way to reproduce so no additional logs are needed from you @marekzebrowski at the moment.

**Comment by patriknw** [comment]
Correction; it stops the negative acks after restarting 2551 (it was just my terminal that was behind). 

Then I'm pretty sure that what is seen here is what was fixed in https://github.com/akka/akka/issues/24847. It actually happened in previous versions also, but then it wasn't noticed because there was no logging. Massive amount of system message resending was going on in the background.

@marekzebrowski Did you try to follow through the full rolling upgrade? You said that a full shutdown was needed. I think it wasn't needed, if you could survive the logging during the rolling upgrade.

**Comment by marekzebrowski** [comment]
I don't have all logs and unfortunately I could not reproduce it - although I tried with apps ran with 2.5.11 and 2.5.12, trying to roll restarts both ways. It seems we could have it only on live :(.
anyways - I managed to get some logs from two systems - maybe they can be helpful.
[logs-to-akka-failed-start.txt](https://github.com/akka/akka/files/2039587/logs-to-akka-failed-start.txt)
Situation from that logs is when we tried to do rolling deployment of application with akka 2.5.12 whereas cluster was running 2.5.11, so it might be one-time issue already fixed in the release. after that problem, restarting all nodes with 2.5.12 made system stable.
Even if it is not reproducible, one thing is noteworthy - we got message:
`a.r.a.SystemMessageAcker: Sending negative acknowledgement of system message [16]`
repeated 2732466 times in logs during that incident (in roughly 2 minutes) - that might be a problem for some logging systems.
